### PR TITLE
Adapt to coq/coq#19620 (Global.push_context_set no strict argument)

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by ppedrot** [comment]
Please merge now.

**Comment by ppedrot** [comment]
Ping @JasonGross

**Comment by JasonGross** [comment]
This is not backwards compatible, it's failing with
```
File "src/Rewriter/Util/plugins/inductive_from_elim.ml", line 48, characters 10-12:
48 |       let () = Global.push_context_set ctx in
               ^^
Error: This pattern should not be a unit literal, the expected type is
       strict:bool -> unit
```
on the Alpine CI.  Are you sure we should be changing the older versions of the code too?

File: src/Rewriter/Util/plugins/inductive_from_elim.ml.v819 [hunk 1]
```diff
@@ -45,7 +45,7 @@ let make_inductive_from_elim sigma (name : Names.Id.t option) (elim_ty : EConstr
     let univs, ubinders = Evd.check_univ_decl ~poly:false sigma UState.default_univ_decl in
     let uctx = match univs with
     | UState.Monomorphic_entry ctx ->
-      let () = Global.push_context_set ~strict:true ctx in
+      let () = Global.push_context_set ctx in
```
Comment (JasonGross) [inline src/Rewriter/Util/plugins/inductive_from_elim.ml.v819:48]:
```suggestion
      let () = Global.push_context_set ~strict:true ctx in
```

File: src/Rewriter/Util/plugins/inductive_from_elim.ml.v820 [hunk 2]
```diff
@@ -45,7 +45,7 @@ let make_inductive_from_elim sigma (name : Names.Id.t option) (elim_ty : EConstr
     let univs, ubinders = Evd.check_univ_decl ~poly:false sigma UState.default_univ_decl in
     let uctx = match univs with
     | UState.Monomorphic_entry ctx ->
-      let () = Global.push_context_set ~strict:true ctx in
+      let () = Global.push_context_set ctx in
```
Comment (JasonGross) [inline src/Rewriter/Util/plugins/inductive_from_elim.ml.v820:48]:
```suggestion
      let () = Global.push_context_set ~strict:true ctx in
```

**Pull request merged**